python-sdk/nuscenes/scripts/export_2d_annotations_as_json.py

```diff
@@ -0,0 +1,137 @@
+
+    for key, value in ann_rec.items():
+        if key in ignored_keys:
+            pass
+        else:
+            repro_rec[key] = value
+
+    repro_rec['bbox_corners'] = [x1, y1, x2, y2]
+    repro_rec['filename'] = filename
+
+    return repro_rec
+
+
+def get_2d_boxes(sample_data_token: str) -> List[OrderedDict]:
+    """
+    Get the 2D annotation records for a given `sample_data_token`.
+    :param sample_data_token: Sample data token belonging to a keyframe.
+    :return: List of 2D annotation record that belongs to the input `sample_data_token`
+    """
+
+    # Get the sample data, and the sample corresponding to that sample data
+    sd_rec = nusc.get('sample_data', sample_data_token)
+    if not (sd_rec['is_key_frame']):
+        raise ValueError("Selected sample data token is not a keyframe")
+    s_rec = nusc.get('sample', sd_rec['sample_token'])
+
+    # Get the calibrated sensor and ego pose record to get the transformation matrices
+    cs_rec = nusc.get('calibrated_sensor', sd_rec['calibrated_sensor_token'])
+    pose_rec = nusc.get('ego_pose', sd_rec['ego_pose_token'])
+    camera_intrinsic = np.array(cs_rec['camera_intrinsic'])
+
+    # Get all the annotation with above a visibility threshold
+    ann_recs = [nusc.get('sample_annotation', token) for token in s_rec['anns']]
+    ann_recs = [ann_rec for ann_rec in ann_recs if (ann_rec['visibility_token'] in args.visibilities)]
+
+    repro_recs = []
+    for ann_rec in ann_recs:
+
+        ann_rec['sample_annotation_token'] = ann_rec['token']
+        ann_rec['sample_data_token'] = sample_data_token
+
+        box = nusc.get_box(ann_rec['token'])
+
+        box.translate(-np.array(pose_rec['translation']))
+        box.rotate(Quaternion(pose_rec['rotation']).inverse)
+
+        box.translate(-np.array(cs_rec['translation']))
+        box.rotate(Quaternion(cs_rec['rotation']).inverse)
+
+        if not box_in_image(box, camera_intrinsic, (1600, 900), 1):
+            continue
+
+        corners = view_points(box.corners(), camera_intrinsic, True)
+        max_x = max(corners[0])
+        min_x = min(corners[0])
+        max_y = max(corners[1])
+        min_y = min(corners[1])
+
+        repro_rec = generate_record(ann_rec, min_x, min_y, max_x, max_y, sample_data_token, sd_rec['filename'])
+        repro_recs.append(repro_rec)
+
+    return repro_recs
+
+
+def main():
+    sample_data_camera_tokens = [s['token'] for s in nusc.sample_data if
+                                 (s['sensor_modality'] == 'camera') and s['is_key_frame']]
+
+    print("Generating 2d reprojections of the nuScenes dataset")
+    reprojections = []
+    for token in tqdm(sample_data_camera_tokens):
+        reprojection_records = get_2d_boxes(token)
+        reprojections.extend(reprojection_records)
+
+    if not os.path.exists(args.dest_path):
+        os.makedirs(args.dest_path)
+
+    with open('{}.json'.format(os.path.join(args.dest_path, args.version)), 'w') as fh:
```

Comment:
I'd suggest you use `args.dataroot + args.version + args.dest_filename` instead. So the user just says what the file name should be and it is stored right in with the other tables. Also set `dest_filename` default to `image_annotations.json`.